.github/workflows/backend-test.yml

```
@@ -0,0 +1,21 @@
+name: backend-test
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  backend-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+
+    - name: Set up Golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```

This can use the matrix referenced above like so:
```suggestion
        go-version: ${{ matrix.go-version }}
```